### PR TITLE
chore: add JSDoc

### PR DESCRIPTION
**Pull request by SamuelAlev**
Add a few word on properties of the sidebar.
In the future, it will have link to the documentation but that is a good start.

### CODE REVIEW

File: packages/sidebar-settings/src/blocks/checkbox.ts [hunk 1]
```diff
@@ -1,9 +1,31 @@
...
+
+    /**
+     * The label for the checkbox.
+     */
     label?: string;
+
+    /**
+     * Additionnal information about the settings to be shown in a tooltip
+     */
+    info?: string;
```
Comment (SamuelAlev) [inline packages/sidebar-settings/src/blocks/checkbox.ts:19]: Moving from `tooltip.content` to `info` to have settings aligned

File: packages/sidebar-settings/src/blocks/fontInput.ts [hunk 2]
```diff
@@ -2,71 +2,264 @@
...
+          fontProvider: Extract<FontProvider, 'Selfhosted'> | `${Extract<FontProvider, 'Selfhosted'>}`;
+
+          /**
+           * The identifier of the font family.
+           */
+          fontFamily: number;
+      }
+) & {
+    /**
+     * The color of the text.
```
Comment (triggertoo) [inline packages/sidebar-settings/src/blocks/fontInput.ts:39]:
```suggestion
     * The rgba-color of the text.
```

File: packages/sidebar-settings/src/blocks/fontInput.ts [hunk 3]
```diff
@@ -2,71 +2,264 @@
...
+     */
     uppercase: boolean;
+
+    /**
+     * Whether the font should be in italic or not.
+     */
     italic: boolean;
+
+    /**
+     * Whether the font should be in underlined or not.
```
Comment (triggertoo) [inline packages/sidebar-settings/src/blocks/fontInput.ts:91]:
```suggestion
     * Whether the font should be underlined or not.
```

File: packages/sidebar-settings/src/blocks/fontInput.ts [hunk 4]
```diff
@@ -2,71 +2,264 @@
...
+     */
     underline: boolean;
 
+    /**
+     * The color of the text on hover.
```
Comment (triggertoo) [inline packages/sidebar-settings/src/blocks/fontInput.ts:96]:
```suggestion
     * The rgba-color of the text on hover.
```

File: packages/sidebar-settings/src/blocks/fontInput.ts [hunk 5]
```diff
@@ -2,71 +2,264 @@
...
+     */
     hoverColor?: {
+        /**
+         * Red color: 0-255
```
Comment (triggertoo) [inline packages/sidebar-settings/src/blocks/fontInput.ts:100]:
```suggestion
         * The red value of the color (0-255).
```

File: packages/sidebar-settings/src/blocks/fontInput.ts [hunk 6]
```diff
@@ -2,71 +2,264 @@
...
+         */
         red: number;
+        /**
+         * Green color: 0-255
```
Comment (triggertoo) [inline packages/sidebar-settings/src/blocks/fontInput.ts:104]:
```suggestion
         * The green value of the color (0-255).
```

File: packages/sidebar-settings/src/blocks/fontInput.ts [hunk 7]
```diff
@@ -2,71 +2,264 @@
...
+         */
         green: number;
+        /**
+         * Blue color: 0-255
```
Comment (triggertoo) [inline packages/sidebar-settings/src/blocks/fontInput.ts:108]:
```suggestion
         * The blue value of the color (0-255).
```

File: packages/sidebar-settings/src/blocks/fontInput.ts [hunk 8]
```diff
@@ -2,71 +2,264 @@
...
+        /**
+         * The green value of the color (0-255).
+         */
         green: number;
+        /**
+         * The blue value of the color (0-255).
+         */
         blue: number;
+        /**
+         * The blue value of the color (0-1).
```
Comment (triggertoo) [inline packages/sidebar-settings/src/blocks/fontInput.ts:55]:
```suggestion
         * The alpha value of the color (0-1).
```

File: packages/sidebar-settings/src/blocks/fontInput.ts [hunk 9]
```diff
@@ -2,71 +2,264 @@
...
+        /**
+         * Green color: 0-255
+         */
         green: number;
+        /**
+         * Blue color: 0-255
+         */
         blue: number;
+        /**
+         * Alpha channel: 0-1
```
Comment (triggertoo) [inline packages/sidebar-settings/src/blocks/fontInput.ts:112]:
```suggestion
         * The alpha value of the color (0-1).
```

File: packages/sidebar-settings/src/blocks/fontInput.ts (hunk 5 above)
Comment (triggertoo) [reply packages/sidebar-settings/src/blocks/fontInput.ts:100]: to be consistent

File: packages/sidebar-settings/src/blocks/fontInput.ts [hunk 10]
```diff
@@ -2,71 +2,264 @@
...
+     */
     activeLineHeight?: string;
+
+    /**
+     * Whether the font should be in uppercase or not when active.
+     */
     activeUppercase: boolean;
+
+    /**
+     * Whether the font should be in italic or not when active.
```
Comment (triggertoo) [inline packages/sidebar-settings/src/blocks/fontInput.ts:244]:
```suggestion
     * Whether the font should be italic or not when active.
```

File: packages/sidebar-settings/src/blocks/fontInput.ts [hunk 11]
```diff
@@ -2,71 +2,264 @@
...
+     */
     activeItalic: boolean;
+
+    /**
+     * Whether the font should be in underlined or not when active.
```
Comment (triggertoo) [inline packages/sidebar-settings/src/blocks/fontInput.ts:249]:
```suggestion
     * Whether the font should be underlined or not when active.
```

File: packages/sidebar-settings/src/bundle.ts [hunk 12]
```diff
@@ -7,7 +7,22 @@ export type SettingValue<AppBridge> = {
...
     getBlock: (id: string) => SettingValue<AppBridge> | null;
+
+    /**
+     * A function that returns an instance of the App Bridge.
+     */
     getAppBridge: () => AppBridge;
-    setBlockValue: (key: string, value: SettingBlock<AppBridge>['value']) => void;
+
+    /**
+     * A function that allows you to programmitically set the value of a given id.
```
Comment (triggertoo) [inline packages/sidebar-settings/src/bundle.ts:23]:
```suggestion
     * A function that allows you to programatically set the value for a block by it's id.
```

File: packages/sidebar-settings/src/bundle.ts [hunk 13]
```diff
@@ -7,7 +7,22 @@ export type SettingValue<AppBridge> = {
...
+     * @param id Represent the setting id.
```
Comment (triggertoo) [inline packages/sidebar-settings/src/bundle.ts:24]:
```suggestion
     * @param id Represents the setting id.
```

**Review by triggertoo — APPROVED** [review]

**Pull request merged**